crates/menmos-client/src/client.rs

```diff
@@ -434,8 +456,8 @@ impl Client {
     /// Pushes a file with the specified meta to the cluster.
     ///
     /// Returns the ID of the created file.
-    pub async fn push<P: AsRef<Path>>(&self, path: P, meta: Meta) -> Result<String> {
-        self.push_internal(path, meta, format!("{}/blob", self.host))
+    pub async fn push<P: AsRef<Path>>(&self, path: P, mut meta: Meta) -> Result<String> {
```

Comment:
Related to my comment above, you can change this to `meta` and pass the meta directly to push_internal instead of a ref - push doesn't need to keep the ownership of the value